k9mail/src/main/java/com/fsck/k9/view/MessageCryptoDisplayStatus.java

```diff
@@ -237,12 +252,15 @@ private static MessageCryptoDisplayStatus getDisplayStatusForPgpResult(CryptoRes
             }
         }
 
+        final long delta = signatureResult.getSignatureTimestamp().getTime() - messageSentDate.getTime();
```

Comment:
If it doesn't have an actual `Date` header it should be considered too old - but we need to check this specifically as K-9 will likely give you the received time if you ask it for `getSentDate()` with no Date header. We may need `isDatePresentOnMessage()` or something to that effect.